### PR TITLE
1554 frontend transaction link summary 

### PR DESCRIPTION
**Pull request by ogerly**
ich habe neu angelegt:

- frontend/src/components/Transactions/TransactionLink.vue
- frontend/src/components/TransactionRows/LinkCountRow.vue
- frontend/src/components/DecayInformations/CollapseLinksList.vue

FireShot Capture 1013 -  - localhost

https://user-images.githubusercontent.com/1324583/158010916-c31ebad6-e60f-404d-b9f0-72b8df838849.mp4

Todo: wir benötigen die anzahl aller aktieven Links im transaktionstype TRANSACTION_LINK

- fixes https://github.com/gradido/gradido/issues/1554

### CODE REVIEW

**Comment by ulfgebhardt** [comment]
use `link45deg` or `link` as icon

image

The icon you currently use is "share"

File: frontend/src/components/DecayInformations/CollapseLinksList.vue [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<template>
+  <div class="collapse-links-list">
+    <div class="d-flex">
+      <div class="text-center pb-3 gradido-max-width">
+        <b-icon icon="droplet-half" height="12" class="mb-2" />
+        <b>{{ $t('links-list.header') }}</b>
```
Comment (ulfgebhardt) [inline frontend/src/components/DecayInformations/CollapseLinksList.vue:6]: if you do the dropdown content in another PR this should open empty or not open at all

```suggestion
```

File: frontend/src/components/GddTransactionList.vue [hunk 2]
```diff
@@ -41,6 +41,10 @@
               :decayStartBlock="decayStartBlock"
             />
           </template>
+
+          <template #TRANSACTION_LINK>
```
Comment (ulfgebhardt) [inline frontend/src/components/GddTransactionList.vue:45]: The name `TRANSACTION_LINK` seems not good, since its more of a `TRANSACTION_LINK_LIST` or `TRANSACTION_LINKS` or something.
Please clarify with @Mogge

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      <div>
+        <b-row>
+          <!-- ICON  -->
+          <b-col cols="1">
+            <type-icon color="text-danger" icon="share" />
+          </b-col>
+
+          <b-col cols="11">
+            <!-- Amount / Name || Text -->
+            <amount-and-name-row :amount="amount" :text="$t('gdd_per_link.links_sum')" />
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:15]: the amount is not negative - clarify with @Mogge 

image

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+            <!-- Count Links -->
+            <link-count-row count="22" />
+
+            <!-- Decay -->
+            <decay-row :decay="decay" />
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:21]: decay is not negative - clarify with @Mogge 

image

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    CollapseIcon,
+    TypeIcon,
+    AmountAndNameRow,
+    LinkCountRow,
+    DecayRow,
+    CollapseLinksList,
+  },
+  props: {
+    amount: {
+      type: String,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:52]: required?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    balance: {
+      type: String,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:55]: not used?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    balanceDate: {
+      type: String,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:58]: not used?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+      type: Object,
+    },
+    id: {
+      type: Number,
+    },
+    linkedUser: {
+      type: Object,
+    },
+    memo: {
+      type: String,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:70]: not used?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    decay: {
+      type: Object,
+    },
+    id: {
+      type: Number,
+    },
+    linkedUser: {
+      type: Object,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:67]: not used?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+      type: String,
+    },
+    balanceDate: {
+      type: String,
+    },
+    decay: {
+      type: Object,
+    },
+    id: {
+      type: Number,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:64]: required?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      type: String,
+    },
+    balance: {
+      type: String,
+    },
+    balanceDate: {
+      type: String,
+    },
+    decay: {
+      type: Object,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:61]: required?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+      type: Number,
+    },
+    linkedUser: {
+      type: Object,
+    },
+    memo: {
+      type: String,
+    },
+    typeId: {
+      type: String,
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:73]: not used?

File: frontend/src/components/Transactions/TransactionLink.vue [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+          <b-col cols="1">
+            <type-icon color="text-danger" icon="share" />
+          </b-col>
+
+          <b-col cols="11">
+            <!-- Amount / Name || Text -->
+            <amount-and-name-row :amount="amount" :text="$t('gdd_per_link.links_sum')" />
+
+            <!-- Count Links -->
+            <link-count-row count="22" />
```
Comment (ulfgebhardt) [inline frontend/src/components/Transactions/TransactionLink.vue:18]: clarify with @Mogge and parameterize

File: frontend/src/locales/de.json [hunk 14]
```diff
@@ -99,6 +99,10 @@
     },
     "your_amount": "Dein Betrag"
   },
+  "gdd_per_link": {
+    "links_count": "aktive Links",
+    "links_sum": "Summe deiner versendeten Gradidos"
```
Comment (ulfgebhardt) [inline frontend/src/locales/de.json:104]: ggf `geblockten Gradidos` - please clarify with @BerndHueckstaedt

**Comment by ulfgebhardt** [comment]
#1594 is not the correct Issue. https://github.com/gradido/gradido/issues/1554 is correct

File: frontend/src/components/DecayInformations/CollapseLinksList.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/components/DecayInformations/CollapseLinksList.vue:6]: i made this yesterday quite quickly and it should only serve as a box where the correct info will be in the future. the icon is not used at this point. but i overlooked it.

File: frontend/src/locales/de.json (hunk 14 above)
Comment (ogerly) [reply frontend/src/locales/de.json:104]: bei @BerndHueckstaedt  angefragt

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 13 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:18]: @Mogge is requested

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 12 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:73]: remove unused

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 11 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:61]: they are all included via v-bind and are necessarily part of the props, as part of the transactions, they are not and should not be passed individually. in none of the tranaction slots is it a requirement that

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 10 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:64]: they are all included via v-bind and are necessarily part of the props, as part of the transactions, they are not and should not be passed individually. in none of the tranaction slots is it a requirement that

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 5 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:52]: they are all included via v-bind and are necessarily part of the props, as part of the transactions, they are not and should not be passed individually. in none of the tranaction slots is it a requirement that

**Comment by ogerly** [comment]
> link45deg

link45deg is used

File: frontend/src/components/GddTransactionList.vue (hunk 2 above)
Comment (Mogge) [reply frontend/src/components/GddTransactionList.vue:45]: Let us discuss how to name it best.

Comment (Mogge) [reply frontend/src/components/GddTransactionList.vue:45]: In this case it is not a list. It is more a summary of all pending transaction links.

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 3 above)
Comment (Mogge) [reply frontend/src/components/Transactions/TransactionLink.vue:15]: You want a negative value to be returned? We should also indicate, that there is included the max decay for each pending transaction link.

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 13 above)
Comment (Mogge) [reply frontend/src/components/Transactions/TransactionLink.vue:18]: OK.

**Review by Mogge — CHANGES_REQUESTED** [review]

File: frontend/src/locales/de.json [hunk 15]
```diff
@@ -101,6 +101,8 @@
     "your_amount": "Dein Betrag"
   },
   "gdd_per_link": {
+    "links_count": "aktive Links",
```
Comment (Mogge) [inline frontend/src/locales/de.json:104]:
```suggestion
    "links_count": "Aktive Links",
```

File: frontend/src/locales/en.json [hunk 16]
```diff
@@ -101,6 +101,8 @@
     "your_amount": "Your amount"
   },
   "gdd_per_link": {
+    "links_count": "active links",
```
Comment (Mogge) [inline frontend/src/locales/en.json:104]:
```suggestion
    "links_count": "Active links",
```

**Review by Mogge — CHANGES_REQUESTED** [review]

File: frontend/src/locales/de.json (hunk 15 above)
Comment (ogerly) [reply frontend/src/locales/de.json]: geändert

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 5 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:52]: erledigt

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 6 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue]: erledigt

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 7 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue]: erledigt

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 9 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue]: erledigt

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 8 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue]: erledigt

File: frontend/src/locales/en.json (hunk 16 above)
Comment (ogerly) [reply frontend/src/locales/en.json]: geändert

File: frontend/src/components/Transactions/TransactionLink.vue (hunk 5 above)
Comment (ogerly) [reply frontend/src/components/Transactions/TransactionLink.vue:52]: erledigt

File: frontend/src/components/GddTransactionList.vue (hunk 2 above)
Comment (ogerly) [reply frontend/src/components/GddTransactionList.vue:45]: If you change the TypeId, then it is virtually only the template in the GddTransactionLIst that is affected.

**Review by Elweyn — APPROVED** [review]
Works on my local machin

**Review by Mogge — APPROVED** [review]

**Review by ulfgebhardt — APPROVED** [review]
💪🏻
